### PR TITLE
[Core] Fix issue with the wrong path of setup logs

### PR DESCRIPTION
**Pull request by Michaelvll**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: sky/backends/cloud_vm_ray_backend.py [hunk 1]
```diff
@@ -3175,9 +3175,17 @@ def _run_setup(setup_cmd: str) -> int:
...
-                with open(setup_log_path, 'r', encoding='utf-8') as f:
-                    if 'too long' in f.read():
-                        is_message_too_long = True
+                try:
+                    with open(os.path.expanduser(setup_log_path),
+                              'r',
+                              encoding='utf-8') as f:
+                        if 'too long' in f.read():
+                            is_message_too_long = True
+                except Exception:  # pylint: disable=broad-except
```
Comment (cblmemo) [inline sky/backends/cloud_vm_ray_backend.py:3184]: Should we except certain error type?

**Review by cblmemo — APPROVED** [review]
Thanks @Michaelvll ! LGTM as a temporary fix in 0.7. More generally, when would this error triggered and should we only except the expected error type?

File: sky/backends/cloud_vm_ray_backend.py (hunk 1 above)
Comment (Michaelvll) [reply sky/backends/cloud_vm_ray_backend.py]: This detection for messages too long is not critical, and we want to make it more conservative here by catching any exceptions. Previously, we got an error for FilesNotFound and the setup crashed.

Comment (Michaelvll) [reply sky/backends/cloud_vm_ray_backend.py]: Added a comment there : )

**Pull request merged**